src/crystal/system/file.cr

```diff
@@ -47,6 +47,43 @@ module Crystal::System::File
     m | o
   end
 
+  LOWER_ALPHANUM = "0123456789abcdefghijklmnopqrstuvwxyz".to_slice
+
+  def self.mktemp(prefix : String?, suffix : String?, dir : String, random : ::Random = ::Random::DEFAULT) : {LibC::Int, String}
+    mode = LibC::O_RDWR | LibC::O_CREAT | LibC::O_EXCL
+    perm = ::File::Permissions.new(0o600)
+
+    prefix = ::File.join(dir, prefix || "")
+
+    100.times do
+      bytesize = prefix.bytesize + 8 + (suffix.try(&.bytesize) || 0)
```

Comment:
This can be computed once outside the loop:

```suggestion
    bytesize = prefix.bytesize + 8 + (suffix.try(&.bytesize) || 0)

    100.times do
```